### PR TITLE
Space layout fix in mobile

### PR DESCRIPTION
**Pull request by ChaituVR**
- Added menu mobile
- also added links to github and twitter

spacelayout

### CODE REVIEW

**Review by samuveth — COMMENTED** [review]
I like the idea, not sure about the increased space and the three dot icon yet. Maybe a burger icon would look good?
image

**Comment by bonustrack** [comment]
Instead of creating SpaceLayout it would be better to setup children routes, this way the menu would remain open when changing page. It would look like this: 
```
{
  path: '/:key',
  component: Space,
  beforeEnter,
  children: [
    { path: '', name: 'space-proposals', component: Proposals },
    { path: 'about', name: 'space-about', component: About }
  ]
}
```
Space would be similar than SpaceLayout but use `<router-view/>` this instead of `</slot>`.

**Comment by ChaituVR** [comment]
@bonustrack Changed that to use router-vie, please review

**Comment by ChaituVR** [comment]
space new

**Comment by ChaituVR** [comment]
[Fix space layout in mobile](https://trello.com/c/nlrZDCGy/137-fix-space-layout-in-mobile)

**Comment by ChaituVR** [comment]
@samuveth @bonustrack plz review when you have time :)

**Review by samuveth — COMMENTED** [review]
Looks great!
Maybe we want to remove the scroll bar?
image

**Comment by ChaituVR** [comment]
@samuveth scrollbar only appears when it is needed 🤔 not shown always..

**Comment by bonustrack** [comment]
@ChaituVR It's never needed

**Comment by ChaituVR** [comment]
@samuveth @bonustrack Scroll bar for menu is hidden now 👍

File: src/components/Block/Space.vue [hunk 1]
```diff
@@ -48,9 +70,7 @@ import { getInstance } from '@snapshot-labs/lock/plugins/vue3';
 import { computed } from 'vue';
 
 export default {
-  props: {
-    space: Object
-  },
+  props: ['space'],
```
Comment (samuveth) [inline src/components/Block/Space.vue:73]: Why are we removing the type check here?

File: src/mixins.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export default {
       return shorten(str);
     },
     _ipfsUrl(ipfsHash: string): string | null {
-      if(!ipfsHash) return null;
+      if (!ipfsHash) return null;
```
Comment (samuveth) [inline src/mixins.ts:41]: I've seen this happen a lot in other PR's, any idea where the syntax without a space is coming from?

File: src/views/Proposals.vue [hunk 3]
```diff
@@ -0,0 +1,118 @@
+<template>
```
Comment (samuveth) [inline src/views/Proposals.vue:1]: Why do we need to create this file for the mobile nav?

File: src/mixins.ts (hunk 2 above)
Comment (ChaituVR) [reply src/mixins.ts:41]: not sure about others but I got it when I run lint 🤔 can remove it?

File: src/views/Proposals.vue (hunk 3 above)
Comment (ChaituVR) [reply src/views/Proposals.vue:1]: No not just for mobile, now that we are using `router-view` we need separate views for `Proposals` and `About`.

File: src/mixins.ts (hunk 2 above)
Comment (samuveth) [reply src/mixins.ts:41]: No, I also have the syntax with the space. Was just wondering why this space keeps getting removed.

File: src/views/Proposals.vue (hunk 3 above)
Comment (samuveth) [reply src/views/Proposals.vue:1]: Oh, that's cool. I've never used a second router-view like that.

**Comment by samuveth** [comment]
@ChaituVR Do you have a way I can test this with a custom domain?

**Comment by bonustrack** [comment]
You can edit domains.json file in local and add "localhost": "yam.eth" to test custom domain

**Comment by bonustrack** [comment]
Please don't merge i'll do a review

**Comment by ChaituVR** [comment]
@samuveth Yes .. tested by changing hosts in the local system..  I was using https://github.com/scottlerch/HostsFileEditor

**Comment by samuveth** [comment]
@ChaituVR can you resolve conflicts?

**Comment by bonustrack** [comment]
- Can you center the avatar of the space, and bring the name under the avatar?
- Also remove the GitHub and Twitter logo, can add it on About page
- Reduce by half the padding on menu item.
- Add same spacing in left than "Proposal".

**Comment by ChaituVR** [comment]
- Added those changes
- Reason why the `Layout` is removed from About.vue is because we don't need it anymore.. we have a new view file now  `src/views/Space.vue` which acts as a layout with `router-view`

**Comment by samuveth** [comment]
When I click "About" while on "Proposals" it jumps and I can no longer see "About" item that I selected. Is there a way we can improve this?
Jun-22-2021 20-08-10

**Comment by bonustrack** [comment]
- When we have a sidebar left, it should automatically create a horizontal menu (example with Timeline)
- Use `Layout` component (and remove the hide classes from sidebar-left) instead of redoing the layout with classes

**Comment by samuveth** [comment]
Works well. I think the timeline needs some more work, it doesn't look to good with "Filters" on the left and the menu centered.

image

**Comment by ChaituVR** [comment]
> Works well. I think the timeline needs some more work, it doesn't look so good with "Filters" on the left and the menu centered.
>

Hmm. Any thoughts on where to place the dropdown?

**Comment by samuveth** [comment]
> > Works well. I think the timeline needs some more work, it doesn't look so good with "Filters" on the left and the menu centered.
> 
> Hmm. Any thoughts on where to place the dropdown?

I mean the "Filters" title

**Comment by ChaituVR** [comment]
Can be centered on both mobile and desktop? or better to center only on mobile 🤔 
image
image

**Comment by ChaituVR** [comment]
Or can we remove "Filters" and add "Timeline" there? 🤔  because in space we have space name in that place

File: src/router.ts [hunk 4]
```diff
@@ -28,7 +29,13 @@ const routes: any[] = [
   {
     path: '/',
     name: 'home',
-    component: domains[domainName] ? Space : Home
+    component: Home,
+    beforeEnter(to, from, next) {
+      if (domains[domainName]) {
```
Comment (bonustrack) [inline src/router.ts:34]: Would be great if there is a way so we don't redirect user from custom domain like vote.balancer.fi to vote.balancer.fi/#/balancer

File: src/views/Timeline.vue [hunk 5]
```diff
@@ -1,22 +1,26 @@
...
-          </div>
-        </Block>
-      </div>
+      <Block
+        :slim="true"
+        :title="$t('filters')"
+        class="text-center overflow-hidden"
+      >
+        <div
+          class="responsivenav pt-2 py-lg-3 px-4 px-lg-0 text-center text-lg-left"
```
Comment (bonustrack) [inline src/views/Timeline.vue:10]: We can align menu in left everywhere, its looks better.

**Pull request closed without merging**